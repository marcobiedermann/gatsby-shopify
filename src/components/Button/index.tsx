import React, { ButtonHTMLAttributes, FC } from 'react';
import * as styles from './style.module.css';

export type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement>;

const Button: FC<ButtonProps> = (props) => {
  return <button className={styles.button} {...props} />;
};

export default Button;
