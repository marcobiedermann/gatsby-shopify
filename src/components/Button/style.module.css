.button {
  background-color: #000;
  color: #fff;
  font-weight: 600;
  padding: 0.75em 1.25em;
}
