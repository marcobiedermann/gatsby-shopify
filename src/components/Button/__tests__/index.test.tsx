import { render } from '@testing-library/react';
import React from 'react';
import Button from '..';

describe('button component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button>Button</Button>);

    expect(asFragment()).toMatchSnapshot();
  });
});
