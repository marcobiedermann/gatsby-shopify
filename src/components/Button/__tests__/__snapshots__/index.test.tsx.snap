// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should render correctly 1`] = `
<button
  class="button"
>
  Button
</button>
`;
